tests/ai_pipeline/test_run.py

```
@@ -0,0 +1,234 @@
+        2. Call run() inside a try/except block.
+        3. Assert no exception is raised; fail the test explicitly if one occurs.
+        """
+        h, w = dummy_bgr_image.shape[:2]
+        pipeline_instance._mock_yolo.return_value = [
+            make_yolo_result(np.array([[0, 0, w, h]], dtype=np.float32))
+        ]
+        pipeline_instance._mock_unet.return_value = make_unet_output()
+
+        try:
+            pipeline_instance.run(str(saved_test_image))
+        except Exception as exc:
```

Ruff (0.15.17)

[warning] Do not catch blind exception: `Exception`

(BLE001)